src/Service/IFido2ServiceFactory.cs

```diff
@@ -0,0 +1,41 @@
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Passwordless.Service.Storage;
+
+namespace Passwordless.Service;
+
+public interface IFido2ServiceFactory
+{
+    Task<IFido2Service> CreateAsync();
+}
+
+public class DefaultFido2ServiceFactory : IFido2ServiceFactory

```

Comment:
Looks good. But if we would refactor the TokenService to not do a Init, we could get rid of this Factory, yes?

Reply:
Yep, and then we just replace injecting the factory and directly inject `IFido2Service`. It's the nicer solution but causes a lot more behavioral changes than I was comfortable doing.